lxd/project/permissions.go

```diff
@@ -152,22 +152,38 @@ func checkRestrictionsOnVolatileConfig(project *api.Project, instanceType instan
 		return nil
 	}
 
+	// List of safe keys
+	safe_keys := [5]string{"volatile.apply_template", "volatile.base_image", "volatile.last_state.power", "volatile.DEVNAME.apply_quota", "volatile.DEVNAME.hwaddr"}
+
 	for key, value := range config {
 		if !strings.HasPrefix(key, shared.ConfigVolatilePrefix) {
 			continue
 		}
 
+		// Allow given safe volatile keys to be set
+		var isSafeKey bool
+		for _, safe_key := range safe_keys {
+			// If current key is in the safe_key list, break out of for loop and set isSafeKey to true
+			if safe_key == key {
+				isSafeKey = true
+				break
+			}
+		}
+
+		// If the current key is a safe volatile key, get out of current iteration
+		if isSafeKey {
+			continue
+		}
+
 		currentValue, ok := currentConfig[key]
 		if !ok {
```

Comment:
From what I can see `checkRestrictionsOnVolatileConfig` is called from two places, `AllowInstanceUpdate` and `AllowInstanceCreation`. From what I understand of @stgraber spec, we should only be silently deleting the non-allowed keys on copy, and not on update.

Also, `AllowInstanceCreation` is called when an instance is being created, but not always from a copy.

See https://github.com/lxc/lxd/blob/d139c8810263788d8596e3b87ffff872c4c311b8/lxd/instances_post.go#L884-L892

So I would expect we would need some sort of extra argument added to this `checkRestrictionsOnVolatileConfig` that would allow us to indicate the type of operation taking place, or perhaps just a boolean that indicates a copy is taking place and we can then alter the behaviour accordingly.

Reply:
@stgraber was it your intention to strip non-allowed keys for all of the creation scenarios above?

Reply:
No, we should only silently drop volatile on `copy` and on `migration`, the other cases would only happen if the user directly fed them to us in which case we should fail.